.github/workflows/firestore.yml

```diff
@@ -316,30 +316,69 @@ jobs:
             --allow-warnings \
             --no-analyze
 
-  spm:
+  spm-source:
     # Either a scheduled run from public repo, or a pull request with firestore changes.
     if: |
       (github.repository == 'Firebase/firebase-ios-sdk' && github.event_name == 'schedule') ||
       (github.event_name == 'pull_request' && needs.changes.outputs.changed == 'true')
     runs-on: macos-12
     needs: check
+    env:
+      FIREBASE_SOURCE_FIRESTORE: 1
     steps:
     - uses: actions/checkout@v3
     - uses: mikehardy/buildcache-action@c87cea0ccd718971d6cc39e672c4f26815b6c126
       with:
         cache_key: ${{ matrix.os }}
     - name: Initialize xcodebuild
       run: scripts/setup_spm_tests.sh
-    - name: iOS Build Test – Binary
+    - name: iOS Build Test
       run: scripts/third_party/travis/retry.sh ./scripts/build.sh FirebaseFirestore iOS spmbuildonly
-    - name: iOS Build Test – Source
+    - name: Swift Build
+      run: scripts/third_party/travis/retry.sh ./scripts/build.sh FirebaseFirestoreSwift iOS spmbuildonly
+
+  spm-binary:
+    # Either a scheduled run from public repo, or a pull request with firestore changes.
+    if: |
+      (github.repository == 'Firebase/firebase-ios-sdk' && github.event_name == 'schedule') ||
+      (github.event_name == 'pull_request' && needs.changes.outputs.changed == 'true')
+    runs-on: macos-12
+    needs: check
+    steps:
+    - uses: actions/checkout@v3
+    - uses: mikehardy/buildcache-action@c87cea0ccd718971d6cc39e672c4f26815b6c126
+      with:
+        cache_key: ${{ matrix.os }}
+    - name: Initialize xcodebuild
+      run: scripts/setup_spm_tests.sh
+    - name: iOS Build Test
       run: scripts/third_party/travis/retry.sh ./scripts/build.sh FirebaseFirestore iOS spmbuildonly
-      env:
-          FIREBASE_SOURCE_FIRESTORE: 1
```

Comment:
Turns out that the build with the env var is leaving behind artifacts that interfere with the next step (which does not use the env var or artifacts from that env var's corresponding build).

Reply:
pre-split results: https://github.com/firebase/firebase-ios-sdk/actions/runs/6197729204/job/16827070395
post-split results: https://github.com/firebase/firebase-ios-sdk/actions/runs/6200662722/job/16835957418 and https://github.com/firebase/firebase-ios-sdk/actions/runs/6200662722/job/16835957681